pkg/serve/server_test.go

```
@@ -183,3 +187,157 @@ func TestServer_MalformedJSON(t *testing.T) {
+	require.NoError(t, err)
+
+	lines := strings.Split(strings.TrimSpace(out.String()), "\n")
+	// ready + at least one result + done
+	require.GreaterOrEqual(t, len(lines), 3)
+
+	var results []Response
+	var done Response
+	for _, line := range lines[1:] {
+		var resp Response
+		require.NoError(t, json.Unmarshal([]byte(line), &resp))
+		if resp.Type == "scan_path_result" {
```

golangci-lint (2.12.2)

[error] QF1003: could use tagged switch on resp.Type

(staticcheck)